server/public/model/remote_cluster.go

```diff
@@ -21,25 +21,30 @@ const (
 	RemoteOfflineAfterMillis = 1000 * 60 * 5 // 5 minutes
 	RemoteNameMinLength      = 1
 	RemoteNameMaxLength      = 64
+
+	BitflagOptionAutoShareDMs Bitmask = 1 << iota
```

Comment:
An options `smallint` using bitflags seems extremely low level -- would it make sense just to add new columns for these options? Or maybe even a JSON blolb? Both would be more accessible to system administrators and developers who might peek directly at the contents of the database. Plus, if we ever need something more than a boolean, we'll need a more flexible data structure anyway.

Reply:
I think this is fine for cases where we have a group of booleans and need to expand them later.  I wish we did more of this in the right circumstances.  We could just keep adding columns, doing the schema changes and migrations as each is needed, but using a JSON blob is the worst of the options IMO.  I agree this is more opaque than well named columns.